modules/virt-configuring-live-migration-heavy.adoc

```diff
@@ -0,0 +1,63 @@
+
+// Module included in the following assemblies:
+//
+// * virt/live_migration/virt-configuring-live-migration.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-configuring-live-migration-heavy_{context}"]
+= Configure live migration for heavy workloads
+
+When migrating a VM running a heavy workload (for example, database processing) with higher memory dirty rates, you need a higher bandwidth to complete the migration.
+
+If the dirty rate is too high, this prevents the migration from one node to another from converging and results in data loss. To prevent this, enable post copy mode.
+
+Post copy mode triggers if the initial pre-copy phase does not complete within the configured timeout parameter. During post copy, the VM CPUs pause on the source host while transfering the minimum required memory pages. Then the VM CPUs activate on the destination host, and the remaining memory pages transfer into the destination node at runtime.
+
+Configure live migration for heavy workloads by updating the `HyperConverged` custom resource (CR), which is located in the `{CNVNamespace}` namespace.
+
+.Procedure
+
+* Edit the `HyperConverged` CR and add the necessary parameters for migrating heavy workloads:
++
+[source,terminal,subs="attributes+"]
+----
+$ oc edit hyperconverged kubevirt-hyperconverged -n {CNVNamespace}
+----
++
+.Example configuration file
+[source,yaml,subs="attributes+"]
+----
+apiVersion: hco.kubevirt.io/v1beta1
+kind: HyperConverged
+metadata:
+  name: kubevirt-hyperconverged
+  namespace: {CNVNamespace}
+spec:
+  liveMigrationConfig:
+    bandwidthPerMigration: 64Mi <1>
+    completionTimeoutPerGiB: 800 <2>
+    parallelMigrationsPerCluster: 5 <3>
+    parallelOutboundMigrationsPerNode: 1 <4>
+    progressTimeout: 150 <5>
+    allowPostCopy: true <6>
+----
+<1> Bandwidth limit of each migration, where the value is the quantity of bytes per second. For example, a value of `2048Mi` means 2048 MiB/s. Default: `0`, which is unlimited.
+//What should this be set to for heavy workloads?
+<2> The migration is canceled if it is not completed in this time, and triggers post copy mode, when post copy is enabled. This value is measured in seconds per GiB of memory. Lower the `completionTimeoutPerGiB` to trigger post copy mode earlier in the migration process.
+//What should it be lowered to?
+<3> Number of migrations running in parallel in the cluster. Default: `5`.
+//Should this be changed for heavy workloads?
+<4> Maximum number of outbound migrations per node. Configure a single VM per node for heavy workloads.
+<5> The migration is canceled if memory copy fails to make progress in this time. This value is measured in seconds. Increase this parameter for large memory sizes running heavy workloads.
+//What is a good amoount to increase this to?
```

Comment:
Question 4

Reply:
It depends on how long admins are willing to wait until a stuck migration fails.
In general, migrations should not get stuck. Nothing getting transferred for 150 seconds (the default) already seems suspicious.
But I guess if the network is really slow and the VMs really busy admins might want to bump that up to something like 15 minutes.

Reply:
Got it. Okay, I don't necessarily think we need to give them a number, in that case.